dipy/viz/ui.py

```diff
@@ -693,3 +700,523 @@ def re_align(self, window_size_change):
+        if vtk.vtkVersion.GetVTKSourceVersion().split(' ')[-1] <= "6.2.0":
+            pass
+        else:
+            text_actor.GetTextProperty().SetBackgroundColor(1, 1, 1)
+            text_actor.GetTextProperty().SetBackgroundOpacity(1.0)
+            text_actor.color(color)
+
+        return text_actor
+
+    def set_message(self, message):
+        """ Set custom text to textbox.
+
+        Parameters
+        ----------
+        message: string
+            The custom message to be set.
+
+        """
+        self.text = message
+        self.actor.set_message(message)
+        self.init = False
+        self.window_right = len(self.text)
+        self.window_left = 0
+        self.caret_pos = self.window_right
+
+    def get_actors(self):
+        """ Returns the actors that compose this UI component. """
+        return [self.actor]
+
+    def add_callback(self, event_type, callback):
+        """ Adds events to the text actor.
+
+        Parameters
+        ----------
+        event_type : string
+            event code
+        callback : string
+            callback function
+
+        """
+        super(TextBox2D, self).add_callback(self.actor, event_type, callback)
+
+    def width_set_text(self, text):
+        """ Adds newlines to text where necessary.
+        This is needed for multi-line text boxes.
+
+        Parameters
+        ----------
+        text : string
+            The final text to be formatted.
+
+        Returns
+        -------
+        multi_line_text : string
+
+        """
+        multi_line_text = ""
+        for i in range(len(text)):
+            multi_line_text += text[i]
+            if (i + 1) % self.width == 0:
+                multi_line_text += "\n"
+        return multi_line_text.rstrip("\n")
+
+    def handle_character(self, character):
+        """ Main driving function that handles button events.
+        # TODO: Need to handle all kinds of characters like !, +, etc.
+
+        Parameters
+        ----------
+        character : string
+
+        """
+        if character.lower() == "return":
+            self.render_text(False)
+            return True
+        if character.lower() == "backspace":
+            self.remove_character()
+        elif character.lower() == "left":
+            self.move_left()
+        elif character.lower() == "right":
+            self.move_right()
+        else:
+            self.add_character(character)
+        self.render_text()
+        return False
+
+    def move_caret_right(self):
+        """ Moves the caret towards right.
+
+        """
+        self.caret_pos += 1
+        if self.caret_pos > len(self.text):
+            self.caret_pos = len(self.text)
+
+    def move_caret_left(self):
+        """ Moves the caret towards left.
+
+        """
+        self.caret_pos -= 1
+        if self.caret_pos < 0:
+            self.caret_pos = 0
+
+    def right_move_right(self):
+        """ Moves right window right.
```

Comment:
The style guide says otherwise: https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#id5